components/MobileNavigation.js

```diff
@@ -0,0 +1,179 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import CSSTransitionGroup from 'react-transition-group/CSSTransitionGroup'
+
+import LevitatingLink from './LevitatingLink'
+import Hamburger from '../resources/hamburger.svg'
+
+export default class MobileNavigation extends Component {
+  static defaultProps = {
+    navs: []
+  }
+
+  static propTypes = {
+    navs: PropTypes.arrayOf(PropTypes.shape({
+      href: PropTypes.string.isRequire,
+      labal: PropTypes.string.isRequire,
+      disabled: PropTypes.bool
+    }))
+  }
+
+  state = {
+    showMobileNav: false,
+    expandMobileNav: false
+  }
+
+  componentDidMount () {
+    window.addEventListener('scroll', this.handleScroll)
+  }
+
+  componentWillUnmount () {
+    window.removeEventListener('scroll', this.handleScroll)
+  }
+
+  handleScroll = (e) => {
+    if (window.scrollY > 700) {
+      if (!this.state.showMobileNav) this.showMobileNav()
+    } else {
+      if (this.state.showMobileNav) this.hideMobileNav()
+    }
+  }
+
+  showMobileNav = () => {
+    this.setState({
+      showMobileNav: true
+    })
+  }
+
+  hideMobileNav = () => {
+    this.setState({
+      showMobileNav: false,
+      expandMobileNav: false
+    })
+  }
+
+  handleToggleMobileNav = (value) => {
+    this.setState({
+      expandMobileNav: value || !this.state.expandMobileNav
+    })
+  }
+
+  handleClickNav = (e) => {
+    this.handleToggleMobileNav(false)
+  }
+
+  render () {
+    const { navs } = this.props
+    const { showMobileNav, expandMobileNav } = this.state
+    return (
+      <CSSTransitionGroup
+        transitionName='fade'
+        transitionEnterTimeout={500}
+        transitionLeaveTimeout={500}>
+        {
+          showMobileNav &&
+            <div id='mobile-nav'>
```

Comment:
This part can be extracted into another function, e.g. `this.renderMobileNav()` so that the `render()` function is not too long and too indented.